### PR TITLE
Add creation parameters to StorageSync ResourceProvider for SDK

### PR DESCRIPTION
**Pull request by ankushbindlish2**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [ ] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/storagesync.json [hunk 1]
```diff
@@ -2508,6 +2535,210 @@
...
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/CloudEndpointCreateParametersProperties",
+          "description": "The parameters used to create the storage sync service."
+        }
+      },
+      "description": "The parameters used when creating a storage sync service."
+    },
+    "CloudEndpointCreateParametersProperties": {
+      "description": "CloudEndpoint Properties object.",
+      "properties": {
```
Comment (marstr) [inline specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/storagesync.json]: Example: This is lacking a lot of the properties that were previously available in "CloudEndpointProperties".

**Review by marstr — CHANGES_REQUESTED** [review]
Changing the names of these types will change the names that are generated. These alone represent substantial breaking changes. Not to mention, in many places (like the one example I've tagged), there are serious differences in what is documented as available via the API.

Almost all of this should be being done in a new API Version.

File: specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/storagesync.json (hunk 1 above)
Comment (ankushbindlish2) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/storagesync.json]: There is NO need of version bump as these changes fixed the swagger spec and make it consistent with what's in the PROD. CSharp verification confirms that.

Also, These changes are aligned with other resource provider has : CreateParameters and UpdateParameters.

Just to bring clarity, There is NO SDK/AzureRM Powershell for existing Swagger spec , which gives us a flexibility to fix the mistakes with in the same version.

Please signoff and unblock us. Hope it helps.

Comment (KrisBash) [reply specification/storagesync/resource-manager/Microsoft.StorageSync/stable/2018-04-02/storagesync.json]: If there are truly no changes to the service behind this, and it is just aligning swagger to the existing API, I will sign-off from ARM's perspective.

**Review by KrisBash — APPROVED** [review]
Signing off from ARM on the condition that this is just aligning swagger to existing API. If there are any corresponding service changes, this should be a new API version/review.

**Review by marstr — APPROVED** [review]
After chatting offline with @ankushbindlish2, we've agreed that these breaking changes are necessary to drive conformance with the behavior of the service. Further more, he's provided evidence of testing these API Specs.

**Pull request merged**